.github/workflows/conformance-gke.yaml

```diff
@@ -119,7 +118,44 @@ jobs:
 
           echo "Generated matrix:"
           cat /tmp/matrix.json
-          echo "matrix=$(jq -c . < /tmp/matrix.json)" >> $GITHUB_OUTPUT
+
+      - name: Set up gcloud credentials
```

Comment:
I wonder if it could make sense to extract the whole matrix generation logic into a separate reusable action, to avoid duplicating it in two different workflows, with the associated divergence issues.

Reply:
I was thinking about it, but the input format of matrixes differs between AKS/EKS/GKE so half of the action like that would be a template substitution specific to cloud-provider. It would be nice to also have `Generate Matrix` and `Convert YAML to JSON` in that action - but they already diverged :shrug: 

So I think it would make sense, but probably as follow-up as we need to clean up other steps first too.

Reply:
Ah, I was actually thinking about a simpler approach that covered only the GKE case.